test/common/TestBedChild.cpp

```diff
@@ -187,10 +211,38 @@ namespace RcclUnitTesting
         }
       }
     }
+
     if (status == TEST_SUCCESS)
-    {
-      CHILD_NCCL_CALL(ncclGroupEnd(), "ncclGroupStart");
+    { 
+      // Check if the communicator is non-blocking
+      if (this->useBlocking == false)
+      { 
+        // handle the ncclGroupEnd in case of non-blocking communication
+        ncclResult_t Group_End_state = ncclGroupEnd();
+        for (int localRank = 0; localRank < numGpus; ++localRank)
+        {
+          ncclResult_t ncclAsyncErrGroupEnd;
+          if (Group_End_state == ncclInProgress) 
+          {
+            do 
+            {
+              ncclCommGetAsyncError(this->comms[localRank], &ncclAsyncErrGroupEnd);
+            } while(ncclAsyncErrGroupEnd == ncclInProgress);
+          }
+          if (ncclAsyncErrGroupEnd != ncclSuccess)
+          {
+            ERROR("Child process %d fails NCCL call %s with code %d\n", this->childId, "ncclCommGetAsyncErrorGroupEnd", ncclAsyncErrGroupEnd);
+            return TEST_FAIL; 
+          }
+        }
+      }
+      else 
+      { 
+        // In case of blocking communication just call ncclGroupEnd
+        CHILD_NCCL_CALL(ncclGroupEnd(), "ncclGroupEnd");
```

Comment:
Perhaps it would make more sense dealing with the blocking vs non-blocking calls by incorporating ncclCommGetAsyncError directly into the CHILD_NCCL_CALL macro